hypothesis-python/tests/cover/test_provisional_strategies.py

```diff
@@ -36,23 +37,18 @@ def test_is_URL(url):
     )
 
 
+@checks_deprecated_behaviour
 @given(ip4_addr_strings())
 def test_is_IP4_addr(address):
     as_num = [int(n) for n in address.split(".")]
     assert len(as_num) == 4
     assert all(0 <= n <= 255 for n in as_num)
 
 
+@checks_deprecated_behaviour
 @given(ip6_addr_strings())
 def test_is_IP6_addr(address):
-    # Works for non-normalised addresses produced by this strategy, but not
-    # a particularly general test
-    assert address == address.upper()
-    as_hex = address.split(":")
-    assert len(as_hex) == 8
-    assert all(len(part) == 4 for part in as_hex)
-    raw = unhexlify(address.replace(":", "").encode("ascii"))
-    assert len(raw) == 16
+    ipaddress.IPv6Address(address)
```

Comment:
Perhaps:

```suggestion
    assert isinstance(ipaddress.IPv6Address(address), str)
```

Reply:
Ah, but it's *not* a string: `address` is a string, but an `IPv6Address` is not.  I've added a comment to this one too, along with an assertion about the type of `address`.